### PR TITLE
Port Gax From Estação Pirata

### PR DESCRIPTION
**Pull request by Ichaie**
# Description

This PR ports the "Gax" map directly from Estação Pirata, making some changes and corrections.
This PR ports https://github.com/rbertoche/space-station-14/pull/80
Gax is a recreated ss13 map, originally from YogStation (SS13)

---

# TODO

---

<details><summary><h1>Media</h1></summary>
<p>

Gax-0

</p>
</details>

---

# Changelog

:cl:
- add: A new map called NCS Gax has been added to rotation.

### CODE REVIEW

File: Resources/Prototypes/Maps/gaxstation.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  mapPath: /Maps/gaxstation.yml
+  minPlayers: 10
+  maxPlayers: 45
+  fallback: true
+  stations:
+    Gaxstation:
+      stationProto: StandardNanotrasenStation
+      components:
+        - type: StationEmergencyShuttle
+          emergencyShuttlePath: /Maps/Shuttles/EstacaoPirata/emergency_neol.yml
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Maps/gaxstation.yml:13]: Im not sure if we have that shuttle file, was it forgotten?

File: Resources/Prototypes/Maps/gaxstation.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        - type: StationNameSetup
+          mapNameTemplate: '{0} NCS Gax {1}'
+          nameGenerator:
+            !type:NanotrasenNameGenerator
+            prefixCreator: '14'
+        - type: StationCargoShuttle
+          path: /Maps/Shuttles/EstacaoPirata/cargo_gax.yml
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Maps/gaxstation.yml:20]: Again i dont think we have this shuttle file.

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]

**Comment by VMSolidus** [comment]
So I don't normally do map reviews, but luckily they aren't that crazy to do since it's pretty self evident if a map is good or not(to me at least), and I have a checklist of things our maps will need. So I'll start off with the Good, the Amazing, the Okay, and the Missing Content.

**Good!**
The layout of this map is awesome, and I appreciate that pretty much every department has their own special interactions for a lot of things. The fully enclosed arrivals dock with meteor shielding is PEAK. I've never seen a map with that before. All of the departments except for Epistemics are incredibly spacious. 

**Awesome:**
I will admit, that in exchange for Epistemics feeling more cramped than the other departments, you have completely knocked it out of the fucking park by futureproofing the map for content that isn't even in the game yet, by including things such as an AI Satellite, a Xenobiology lab, an AI Upload Room(Separate from the Satellite), a fully stocked Robotics Bay with Surgical Room, Server room completely separate from the Mystagogue's office is great. 

You have probably the best medbay I've ever seen in a map. Congratulations on absolutely killing it here. Having Virology, Genetics, Security Checkpoint, Chemistry, Cloning Lab, Cryo bay, Surgery Room, and hell there's even a break room, and a Paramedic checkpoint! There is very little this medbay is missing!

**Okay:** 
Security can stand to be improved most, but luckily its improvements aren't that crazy. I absolutely love that Security has a shuttle dock, presumably in preparation for things like Lavaland Deportation. That's awesome!

The Permabrig could probably be livened up a little, it's unusually barebones I will admit.
image

**Armory is unusually vulnerable:** Lastly, the Armory has a potential access route that would allow for breaching the armory to space via only a single C4. There should be an extra layer of walls on the outside of the armory, which will prevent the armory from being entered with only a single C4. It's important for game balance that Security have time to react when they first hear the beeping that indicates Nukies are trying to take the Armory.
image

**Missing Content:**
Okay, here's where the nit and gritty needs to happen.

**Missing Oracle**: We require that the entity called Oracle(A statue usually paired with Sophia) be placed somewhere on the map, preferably in the Science/Epistemics lobby.
image

**Missing Biomass Reclaimer Drain**: The biomass reclaimer is in an odd spot, placed inside the cloning bay. Usually we'd see it tucked away in the morgue. But I'm not going to ask you to move the reclaimer all the way to the morgue seeing as it's on the exact opposite side of the medbay. It'll just need to have a Drain placed under or in front of it.

**Missing Forensic Mantis:** We require a space be added for the ForensicMantis role be placed somewhere in the map, as well as a Job Spawner for the role. Given the map's unusual layout, you will probably want to convert one of the Maintenance Rooms into a sort of office/drugden for the Mantis to spawn in. Either of these two rooms are suitable(Although the second room may need to be expanded 1 or 2 tiles down):
image

image

**Science-Security Checkpoint is Inaccessible:** This security checkpoint into Science cannot be reached by Security officers. This could be rectified by adding a Security airlock door leading into Maintenance.
image

**Artifact Lab is missing Wall-Mounted Geiger Counters:**
image

File: Content.IntegrationTests/Tests/PostMapInitTest.cs [hunk 3]
```diff
@@ -61,7 +61,8 @@ public sealed class PostMapInitTest
             "TheHive", //DeltaV
             "Hammurabi", //DeltaV
             "Lighthouse", //DeltaV
-			"Submarine" //DeltaV
+			"Submarine", //DeltaV
```
Comment (DEATHB4DEFEAT) [inline Content.IntegrationTests/Tests/PostMapInitTest.cs:64]:
```suggestion
            "Submarine", //DeltaV
```

File: Resources/Prototypes/Maps/gaxstation.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            Librarian: [ 1, 1 ]
+            #Security
+            HeadOfSecurity: [ 1, 1 ]
+            Warden: [ 1, 1 ]
+            SecurityOfficer: [ 4, 4 ]
+            SecurityCadet: [ 4, 4 ]
+            Detective: [ 1, 1 ]
+            Prisoner: [ 1, 2 ]
+            PrisonGuard: [ 1, 2 ]
+            Lawyer: [ 1, 1 ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/Maps/gaxstation.yml:59]:
```suggestion
            Lawyer: [ 1, 1 ]
```

File: Resources/Prototypes/Maps/gaxstation.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            #Security
+            HeadOfSecurity: [ 1, 1 ]
+            Warden: [ 1, 1 ]
+            SecurityOfficer: [ 4, 4 ]
+            SecurityCadet: [ 4, 4 ]
+            Detective: [ 1, 1 ]
+            Prisoner: [ 1, 2 ]
+            PrisonGuard: [ 1, 2 ]
+            Lawyer: [ 1, 1 ]            
+            #Supply
```
Comment (VMSolidus) [inline Resources/Prototypes/Maps/gaxstation.yml:60]:
```suggestion
            PrisonGuard: [ 1, 2 ]
            Brigmedic: [ 1, 1 ]
            Lawyer: [ 1, 1 ]            
            #Supply
```
You also need to add a spawner for said Brigmedic(Localized as Corpsman in our repo)

File: Resources/Prototypes/Maps/gaxstation.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            MedicalBorg: [ 1, 1 ]
+            #Epistemics
+            ResearchDirector: [ 1, 1 ]
+            Scientist: [ 4, 4 ]
+            ResearchAssistant: [ 2, 2 ]
+            Roboticist: [ 1, 2 ]
+            Borg: [ 2, 2 ]
+            Chaplain: [ 1, 1 ]
+            Librarian: [ 1, 1 ]
+            #Security
```
Comment (VMSolidus) [inline Resources/Prototypes/Maps/gaxstation.yml:51]:
```suggestion
            Chaplain: [ 1, 1 ]
            Librarian: [ 1, 1 ]
            ForensicMantis: [ 1, 1 ]
            #Security
```
Will also need a spawner added.

**Review by VMSolidus — CHANGES_REQUESTED** [review]

**Comment by rbertoche** [comment]
Can't miss the opportunity to say the map is awesome though, me myself and players in general like it! It's surely been a milestone on original maps on Pirata, thanks mostly to @Ichaie, also others but I think it's mostly Ichai's.

**Comment by VMSolidus** [comment]
In any case, the changes I've requested are small and easy to do, so I don't mind approving this after they're done. :)

**Comment by VMSolidus** [comment]
Also the map yml has some "INVALID" entries in it. Those will need to be deleted.

**Comment by Ichaie** [comment]
imagem_2024-10-08_200311191
imagem_2024-10-08_200347614
image
@VMSolidus Here are some images of the changes to the map

**Comment by VMSolidus** [comment]
Oh my gosh that Permabrig is gorgeous now.

**Comment by VMSolidus** [comment]
image

Looks like it needs to have a spawn point added for the medical borg. But all the rest of the tests are fine. :)

**Review by VMSolidus — APPROVED** [review]

**Review by FoxxoTrystan — APPROVED** [review]

**Pull request merged**